opal/mca/btl/base/btl_base_select.c

```diff
@@ -107,6 +122,16 @@ int mca_btl_base_select(bool enable_progress_threads,
                                 "select: no init function; ignoring component %s",
                                 component->btl_version.mca_component_name);
         } else {
+#if MEMORY_LINUX_MALLOC_ALIGN_ENABLED
+            if (strcmp(component->btl_version.mca_component_name, "openib") == 0) {
```

Comment:
Fair enough.  Perhaps this could be exposed as a function pointer on the module -- hence, the solaris module can provide a NULL pointer, and the linux one can provide a non-NULL pointer.